### PR TITLE
skeleton global planner

### PR DESCRIPTION
**Pull request by ashwin2802**

### CODE REVIEW

**Review by kpiyush25 — DISMISSED** [review]

File: voxblox_global_planner/src/skeleton_global_planner.cpp [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  skeleton_generator_.generateSkeleton();
+  generateSparseGraph();
+  if (verbose_) {
+    ROS_INFO("Finished generating sparse graph.");
+    ROS_INFO_STREAM("Total Timings: " << std::endl
+                                      << voxblox::timing::Timing::Print());
+  }
+
+    if (getMapDistance(goal_pose.position_W) < constraints_.robot_radius)
+    {
```
Comment (amartyadash) [inline voxblox_global_planner/src/skeleton_global_planner.cpp]: Bracketing convention not followed here, and in other if and for loops

File: voxblox_global_planner/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# LIBRARIES #
+#############
+cs_add_library(${PROJECT_NAME}
+  src/skeleton_global_planner.cpp
+)
+
+############
+# BINARIES #
+############
+
```
Comment (gajena) [inline voxblox_global_planner/CMakeLists.txt]: remove this

File: voxblox_global_planner/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+cs_add_executable(skeleton_global_planner_node
+  src/skeleton_global_planner_node.cpp
+)
+target_link_libraries(skeleton_global_planner_node ${PROJECT_NAME})
+
+##########
+# EXPORT #
+##########
+
```
Comment (gajena) [inline voxblox_global_planner/CMakeLists.txt]: remove this

File: voxblox_global_planner/src/skeleton_global_planner.cpp [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    bool shorten_graph = true;
+
+    voxblox::AlignedVector<voxblox::Point> diagram_coordinate_path;
+    mav_trajectory_generation::timing::Timer astar_diag_timer(
+        "plan/astar_diag");
+    bool success = skeleton_planner_.getPathUsingEsdfAndDiagram(
+        start_point, goal_point, &diagram_coordinate_path);
+    mav_msgs::EigenTrajectoryPointVector diagram_path;
+    for (const voxblox::Point &voxblox_point : diagram_coordinate_path)
+    {
```
Comment (gajena) [inline voxblox_global_planner/src/skeleton_global_planner.cpp]: braces

File: voxblox_global_planner/src/skeleton_global_planner.cpp [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+      mav_msgs::EigenTrajectoryPoint point;
+      point.position_W = voxblox_point.cast<double>();
+      diagram_path.push_back(point);
+    }
+    double path_length = mav_planning::computePathLength(diagram_path);
+    int num_vertices = diagram_path.size();
+    astar_diag_timer.Stop();
+
+    if (visualize_)
+    {
```
Comment (gajena) [inline voxblox_global_planner/src/skeleton_global_planner.cpp]: same

File: voxblox_global_planner/src/skeleton_global_planner.cpp [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+      marker_array.markers.push_back(mav_planning::createMarkerForPath(
+          diagram_path, frame_id_, mav_visualization::Color::Purple(),
+          "astar_diag", 0.1));
+    }
+    ROS_INFO("Diag A* Success? %d Path length: %f Vertices: %d", success,
+             path_length, num_vertices);
+    
+    if (success && getMapDistance(start_pose.position_W) < constraints_.robot_radius)
+    {
```
Comment (gajena) [inline voxblox_global_planner/src/skeleton_global_planner.cpp]: same

File: voxblox_global_planner/src/skeleton_global_planner.cpp [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+      ROS_ERROR("Start pose occupied!");
+      return false;
+    }
+
+    if (shorten_graph)
+    {
```
Comment (gajena) [inline voxblox_global_planner/src/skeleton_global_planner.cpp]: same

**Review by Divyanshu23 — DISMISSED** [review]

File: frontier_explorer/include/frontier_explorer/frontier_evaluator.hpp [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+        bool accurate_frontiers_;
+        bool visualize_;    
+
+        std::string frame_id_;
+
+        ros::Publisher frontier_pub_;
+        ros::Publisher voxel_pub_;
+};
+
```
Comment (gajena) [inline frontier_explorer/include/frontier_explorer/frontier_evaluator.hpp:50]: remove this/

File: voxblox_global_planner/src/skeleton_global_planner.cpp (hunk 1 above)
Comment (ashwin2802) [reply voxblox_global_planner/src/skeleton_global_planner.cpp]: :'(

**Review by gajena — APPROVED** [review]

**Review by amartyadash — APPROVED** [review]

**Pull request merged**